delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/client/SparqlQuery.scala

```diff
@@ -3,18 +3,56 @@ package ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.client
 import akka.http.scaladsl.model.MediaTypes
 import akka.http.scaladsl.unmarshalling.{FromEntityUnmarshaller, FromStringUnmarshaller, PredefinedFromEntityUnmarshallers, Unmarshaller}
 import ch.epfl.bluebrain.nexus.delta.rdf.RdfMediaTypes
+import cats.syntax.all._
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.client.SparqlQuery.SparqlConstructQuery
+import org.apache.jena.query.QueryFactory
 
-trait SparqlQuery {
+import scala.util.Try
+
+sealed trait SparqlQuery {
 
   /**
     * @return string representation of the query
     */
   def value: String
+
+  /**
+    * @return the construct query, if available
+    */
+  def asConstruct: Option[SparqlConstructQuery]
 }
 
 object SparqlQuery {
 
-  final private case class AnySparqlQuery(value: String) extends SparqlQuery
+  /**
+    * Any Sparql query
+    */
+  final private case class AnySparqlQuery(value: String) extends SparqlQuery {
+    override def asConstruct: Option[SparqlConstructQuery] = SparqlConstructQuery(value).toOption
+  }
+
+  /**
+    * Sparql construct query representation.
+    *
+    * @param value  string representation of the query
+    */
+  final case class SparqlConstructQuery private (value: String) extends SparqlQuery {
```

Comment:
There are also DESCRIBE and ASK queries that we have to consider. The DESCRIBE query is similar to the construct one in terms of response type.

Reply:
I'm not considering this yet.

Reply:
Ok. I'll adapt the PR